src/dict.h

```diff
@@ -45,6 +45,12 @@ typedef struct dictType {
     /* Invoked at the end of dict initialization/rehashing of all the entries from old to new ht. Both ht still exists
      * and are cleaned up after this callback.  */
     void (*rehashingCompleted)(dict *d);
+    /* Invoked when the size of the dictionary changes.
+     * - Adds dict[1]'s size at the start of rehashing.
+     * - Subtracts dict[0]'s size at the end of rehashing.
+     * - Subtracts both dict[0] and dict[1] sizes when the dictionary is emptied or released.
+     * The `delta` parameter can be positive (size increase) or negative (size decrease). */
+    void (*sizeChanged)(dict *d, long long delta);
```

Comment:
not sure it's a good name, maybe `dictSizeChanged`.

Reply:
I prefer `bucketChanged`, I `size` feels like the `used` in dict, and p.s. we have `dictSize` and `dictBuckets`